sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1223,6 +1223,12 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       case c: CreateNamedStruct => c.valExprs
       case other => Seq(other)
     }
+    // Check the argument of boolean test is valid.
+    def checkBooleanTestArgs(e: Expression): Unit = e.dataType match {
```

Comment:
Since this is used only here, `private def`.

Reply:
OK

Reply:
Oh. My bad. This is already inside a function.